### ISSUE TITLE
Prosody Encoder

### ISSUE THREAD

**Issue by LiangTing1** [sh-lee-prml/HierSpeechpp#35]
Hi, Is the output of the prosody encoder in Hierarchical speech synthesizer only used to calculate the loss with the first 20 dimensions of the target mel? What is the weight assigned to this loss?

**Comment by sh-lee-prml** [comment]
Hi

Yes. we used the first 20 dimensions of the target mel, and we used the weight of 45 for this loss.

```
        loss_prosody = (torch.sum(torch.abs(mel[:,:hps.model.prosody_size,:] - prosody_hat.float())*mask) / (torch.sum(mask) * hps.model.prosody_size)) * hps.train.c_mel
```

Thanks!

**Comment by LiangTing1** [comment]
Tanks very much for your response.
Would it be possible for you to show the loss curve?
this is my training loss , the sixth is prosody encoder loss, Does this value seem a bit large?
<img width="1439" alt="image" src="https://github.com/sh-lee-prml/HierSpeechpp/assets/127732298/6336f58c-cc8b-46e0-920b-a0dec049980c">

**Comment by sh-lee-prml** [comment]
Hi 

Here is our tensorboard logs

![image](https://github.com/sh-lee-prml/HierSpeechpp/assets/56749640/2f88b825-984a-4e31-b4db-239cd679075a)

green is from the scratch with LibriTTS 460, and orange is from the green with Full-dataset.

**Comment by hoangtm-aimesoft** [comment]
> Tanks very much for your response. Would it be possible for you to show the loss curve? this is my training loss , the sixth is prosody encoder loss, Does this value seem a bit large? <img alt="image" width="1439" src="https://private-user-images.githubusercontent.com/127732298/302098276-6336f58c-cc8b-46e0-920b-a0dec049980c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyODczMjQsIm5iZiI6MTcwOTI4NzAyNCwicGF0aCI6Ii8xMjc3MzIyOTgvMzAyMDk4Mjc2LTYzMzZmNThjLWNjOGItNDZlMC05MjBiLWEwZGVjMDQ5OTgwYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwMVQwOTU3MDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MGE0MDgwZjk4NTNiYjg3NzQyNjQ5YWU0ZDUxODRlMDQxYzFlNDgwZThkZGZjNTE3MTRkYWFiMDEwNDI2ZGVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.5Y9RK_plgMoV8HCLNuq9Ihjpgv5ZLhM3fdvWVU3RYgM">
Can you kindly share the training code for the hierachical speech synthesizer. Thank you very much !

**Comment by hoangtm-aimesoft** [comment]
> Tanks very much for your response. Would it be possible for you to show the loss curve? this is my training loss , the sixth is prosody encoder loss, Does this value seem a bit large? <img alt="image" width="1439" src="https://private-user-images.githubusercontent.com/127732298/302098276-6336f58c-cc8b-46e0-920b-a0dec049980c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0MDA2MzAsIm5iZiI6MTcxMDQwMDMzMCwicGF0aCI6Ii8xMjc3MzIyOTgvMzAyMDk4Mjc2LTYzMzZmNThjLWNjOGItNDZlMC05MjBiLWEwZGVjMDQ5OTgwYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxNFQwNzEyMTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDhhMDdlODEwZWM2ZGY5YmY0OWQyYTdjNDQ5MmYzYjI1YmJlYzBlN2UyNzA0M2RkNjE1MmIyYWZhYzE4ZjVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jyRid-Zbwyx1qpoCS7IWhDv5eUZrtLkR6Qnni4pc-9w">

Can you kindly share you training code for the hierarchical speech synthesizer ?
